pkg/metrics/firewall.go

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: MIT
+
+package metrics
+
+import (
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/extensions/fwaas/firewalls"
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/klog"
+)
+
+var (
+	firewallAdminStateUp *prometheus.GaugeVec
+	firewallStatus       *prometheus.GaugeVec
+
+	// according to the nl_constants from https://github.com/openstack/neutron-fwaas/blob/stable/ocata/neutron_fwaas/services/firewall/fwaas_plugin.py
+	// we will get the following firewall states
+	firewallStates = []string{"ACTIVE", "DOWN", "ERROR", "INACTIVE", "PENDING_CREATE", "PENDING_UPDATE", "PENDING_DELETE"}
+
+	firewallLabels = []string{"id", "name", "description", "policyID", "projectID"}
+)
+
+func registerFirewallMetrics() {
+	firewallAdminStateUp = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: generateName("firewall_admin_state_up"),
```

Comment:
```suggestion
			Name: generateName("firewall_v1_admin_state_up"),
```
also for the others pls